src/firebase/auth.ts

```diff
@@ -36,3 +42,28 @@ export const signOutUser = async () => {
     console.log(error);
   }
 };
+
+export const registerWithEmailAndPassword = async (
```

Comment:
for these two functions (registerWithEmailAndPassword and signInWithEmailAndPassword) you probably don't need to make separate back end functions in auth.ts since they just call one function. you can just call them in the screens when buttons get pressed. if at some point you want to bundle up more back end functionality related to registering/signing up with email and password then it's probably good to create separate functions in auth.ts to make things cleaner